app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,43 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    if user_id = session[:user_id]
+      @current_user ||= User.find_by id: user_id
+    elsif user_id = cookies.signed[:user_id]
+      user = User.find_by id: user_id
+      if user&.authenticated? cookies[:remember_token]
+        log_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def logged_in?
+    current_user.present?
+  end
+
+  def log_out
+    forget current_user
+    session.delete :user_id
+    @current_user = nil
+  end
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
+  def admin_login?
+    current_user.present? && current_user.role == 1
```

Comment:
```suggestion
    current_user.present? && current_user.admin?
```

việc check user đã login hay chưa tách hãng 2 method và có message khi user chưa login nhé, để tái sử dụng

ở đây suggest dùng lần lượt đầu tiên phải login -> nếu chưa báo lỗi
login ko pahir admin --> báo lỗi

Reply:
Method admin_login? em viết để hiển thị giao diện thôi ạ, còn việc kiểm tra quyền admin thì em đã tạo admin base controller ở phần sau để check quyền admin trước khi thực hiện các method và điều hướng trang đến đăng nhập nếu không phải admin ạ.